### ISSUE TITLE
ephoto deletes files when it cannot save

### ISSUE THREAD

**Issue by wltjr** [okratitan/Ephoto#1]
I have encountered this once before. I am now looking at an image in ephoto it deleted from disk. It had a problem saving, and as a result. It deleted the original. This is not the first time that happened. Its making me very afraid to even use it. Since it keeps deleting the original file. Really need to see about doing something differently. It should never delete the original. If there is any error during a save it should not effect the original. This also happens if there is an error and it crashes. Quite frustrating....

**Comment by okratitan** [comment]
Interesting. First I've heard of that but I can see where and why that happens. You generally assume that if you have permission to unlink a file you would have permission to create one, which is basically what save does, unlink the old and save the new. There is a backup mechanism already in place that I can add to saving files.

**Comment by wltjr** [comment]
Yeah not sure, the best way maybe copy original to ~ or /tmp till the new one is written. Then on fall back go to the other. Or what ever you have in place :) I should have mentioned it before. Both times were related to some sort of change. This time just rotate. Stupid phones always rotate images... :|

**Comment by wltjr** [comment]
Great thank you very much! Any chance your doing a new release soon? Any other changes worth such? If not I can make a tagged version. No rush I am happy to wait a bit. Just tend to prefer tags. I have scripts to bump/update ebuilds/packages. Snapshots are harder to deal with. I also have live, but its not smart, doesn't even know if any changes were made in git. So it would just checkout and rebuild all the time and no easy way to do that for many packages... :) As the backstory as to why tag etc. Either way thanks again!

**Comment by okratitan** [comment]
I will be doing another release before year end.

Cheers, and as always, keep reporting your findings.

**Comment by wltjr** [comment]
Around end or start of year is great, thanks! I will report issues considering I can here :P I really haven't had any issues. There is one I wanted to bring up. More a design thing will file on that. If I can remember or replicate. It was something like going into a folder and getting stuck not being able to go back to parent. I am not able to replicate its been a bit. That was pretty minor. But if I have any issues will let you know. Thanks again!
